core/client/fs/src/test/java/alluxio/client/file/FileSystemContextTest.java

```diff
@@ -66,6 +70,37 @@ public void acquireAtMaxLimit() throws Exception {
     }
   }
 
+  @Test
+  public void getCache() throws Exception {
+    FileSystemContext ctx1 = FileSystemContext.get();
+    FileSystemContext ctx2 = FileSystemContext.get();
+    assertEquals(ctx1, ctx2);
+  }
+
+  @Test
+  public void getDifferentSubjects() throws Exception {
+    Subject sub = new Subject();
+    FileSystemContext ctx1 = FileSystemContext.get();
+    FileSystemContext ctx2 = FileSystemContext.get(sub);
+    FileSystemContext ctx3 = FileSystemContext.get(sub);
+    assertNotEquals(ctx1, ctx2);
+    assertEquals(ctx2, ctx3);
+  }
+
+  @Test
+  public void refCount() throws Exception {
+    FileSystemContext ctx1 = FileSystemContext.get();
+    FileSystemContext ctx2 = FileSystemContext.get();
+    ctx1.close();
+    FileSystemContext ctx3 = FileSystemContext.get();
+    assertEquals(ctx1, ctx3); // Same context
```

Comment:
there's `assertSame`/`assertNotSame` for checking if the arguments are references to the same object

Reply:
Good catch, updated